app/src/main/java/pm/gnosis/heimdall/ui/addressbook/edit/AddressBookEditEntryActivity.kt

```diff
@@ -0,0 +1,127 @@
+import pm.gnosis.svalinn.common.utils.snackbar
+import pm.gnosis.svalinn.common.utils.subscribeForResult
+import pm.gnosis.svalinn.common.utils.toast
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+import pm.gnosis.utils.exceptions.InvalidAddressException
+import timber.log.Timber
+import javax.inject.Inject
+
+class AddressBookEditEntryActivity : ViewModelActivity<AddressBookContract>() {
+    @Inject
+    lateinit var toolbarHelper: ToolbarHelper
+
+    override fun layout() = R.layout.layout_address_book_update_entry
+
+    override fun inject(component: ViewComponent) = component.inject(this)
+
+    override fun screenId() = ScreenId.ADDRESS_BOOK_EDIT_ENTRY
+
+    private lateinit var address: Solidity.Address
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        intent.getStringExtra(EXTRA_ADDRESS).let {
+            it.asEthereumAddress()?.let {
+                address = it
+            } ?: run {
+                toast(R.string.invalid_ethereum_address)
+                finish()
+                return
+            }
+        }
+
+        // Needs to be done on onCreate because the result of the QR Scan activity would get "overridden" by the DB load
+        disposables += viewModel.loadAddressBookEntry(address)
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeBy(onSuccess = {
+                layout_address_book_update_entry_name.setText(it.name)
+                layout_address_book_update_entry_address.setText(it.address.asEthereumAddressString())
+            }, onError = {
+                Timber.e(it)
+                finish()
+            })
+    }
+
+    override fun onStart() {
+        super.onStart()
+        disposables += toolbarHelper.setupShadow(layout_address_book_update_entry_toolbar_shadow, layout_address_book_update_entry_scroll_view)
+
+        disposables += layout_address_book_update_entry_scan.clicks()
+            .subscribeBy(onNext = { QRCodeScanActivity.startForResult(this) })
+
+        disposables += layout_address_book_update_entry_save.clicks()
+            .flatMap {
+                viewModel.updateAddressBookEntry(
+                    address,
+                    layout_address_book_update_entry_name.text.toString(),
+                    layout_address_book_update_entry_address.text.toString()
+                )
+            }
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeForResult(onNext = ::onAddressBookEntryUpdated, onError = ::onAddressBookEntryUpdateError)
+
+        disposables += layout_address_book_update_entry_back_arrow.clicks()
+            .subscribeBy(onNext = { onBackPressed() }, onError = Timber::e)
+    }
+
+
+    private fun onAddressBookEntryUpdated(entry: AddressBookEntry) {
+        startActivity(AddressBookActivity.createIntent(this).apply { flags = Intent.FLAG_ACTIVITY_CLEAR_TOP })
```

Comment:
Why do we start the address book activity and not just finish to go back to the previous, screen?